eventio/base.py

```diff
@@ -18,18 +19,66 @@
 KNOWN_OBJECTS = {}
 
 
+class PipeWrapper:
+    '''
+    This class makes a sp.PIPE  forward-seekable
+    by keeping track of the bytes already read
+    and reading as many bytes as required in `seek`
+    '''
+    def __init__(self, pipe):
+        self.pipe = pipe
+        self.pos = 0
+
+    def read(self, size=-1):
+        data = self.pipe.read(size)
+        self.pos += len(data)
+        return data
+
+    def seek(self, offset, whence=0):
+        if whence == 0:
+            to_read = offset - self.pos
+            if to_read < 0:
+                raise IOError('Only forward seeking possible')
+            self.pos += len(self.pipe.read(to_read))
+
+        if whence == 1:
+            if offset < 0:
+                raise IOError('Only forward seeking possible')
+            self.pos += len(self.pipe.read(offset))
+
+        if whence == 2:
+            raise IOError('Only forward seeking possible')
+
+        return self.pos
+
+    def close(self):
+        return self.pipe.close()
+
+
 class EventIOFile:
 
-    def __init__(self, path):
+    def __init__(self, path, zcat=True):
         log.info('Opening new file {}'.format(path))
         self.path = path
+        self.read_process = None
 
         if not is_eventio(path):
             raise ValueError('File {} is not an eventio file'.format(path))
 
         if is_gzip(path):
             log.info('Found gzipped file')
-            self._filehandle = gzip.open(path, mode='rb')
+            if zcat:
+                try:
+                    log.info('Trying to read using zcat')
+                    self.read_process = sp.Popen(['zcat', path], stdout=sp.PIPE)
+                    self._filehandle = PipeWrapper(self.read_process.stdout)
+                    log.info('Using zcat')
+                except Exception:
+                    log.info('Falling back to gzip module')
```

Comment:
I think this might be a warning ... since something went wrong...  and all these info logs ... I never get when to use info got debug...